content/blog/2025-03-20-datafusion-comet-0.7.0.md

```diff
@@ -0,0 +1,134 @@
+contributors. See the [change log] for more information.
+
+[change log]: https://github.com/apache/datafusion-comet/blob/main/dev/changelog/0.7.0.md
+
+## Release Highlights
+
+### Performance
+
+Comet 0.7.0 has improved performance compared to the previous release due to improvements in the native shuffle 
+implementation and performance improvements in DataFusion 46.
+
+For single-node TPC-H at 100 GB, Comet now delivers a **greater than 2x speedup** compared to Spark using the same 
+CPU and RAM. Even with **half the resources**, Comet still provides a measurable performance improvement.
+
+<img
+src="/blog/images/comet-0.7.0/performance.png"
+width="100%"
+class="img-responsive"
+alt="Chart showing TPC-H benchmark results for Comet 0.7.0"
+/>
+
+*These benchmarks were performed on a Linux workstation with PCIe 5, AMD 7950X CPU (16 cores), 128 GB RAM, and data 
+stored locally in Parquet format on NVMe storage. Spark was running in Kubernetes with hard memory limits.*
+
+## Shuffle Improvements
+
+There are several improvements to shuffle in this release:
+
+- When running in off-heap mode (which is the recommended approach), Comet was using the wrong memory allocator 
+  implementation for some types of shuffle operation, which could result in OOM rather than spilling to disk.
+- The number of spill files is drastically reduced. In previous releases, each instance of ShuffleMapTask could 
+  potentially create a new spill file for each output partition each time that spill was invoked. Comet now creates 
+  a maximum of one spill file per output partition per instance of ShuffleMapTask, which is appended to in subsequent 
+  spills.
+- There was a flaw in the memory accounting which resulted in Comet requesting approximately twice the amount of 
+  memory that was needed, resulting in premature spilling. This is now resolved.
+- The metric for number of spilled bytes is now accurate. It was previously reporting invalid information.
+
+## Improved Hash Join Performance
+
+When using the `spark.comet.exec.replaceSortMergeJoin` setting to replace sort-merge joins with hash joins, Comet 
+will now do a better job of picking the optimal build side. Thanks to [@hayman42] for suggesting this, and thanks to the 
+[Apache Gluten(incubating)] project for the inspiration in implementing this feature.
+
+[@hayman42]: https://github.com/hayman42
+[Apache Gluten(incubating)]: https://github.com/apache/incubator-gluten/
+
+## Experimental Support for DataFusion’s Parquet Scan
+
+It is now possible to configure Comet to use DataFusion’s Parquet reader instead of Comet’s current Parquet reader. This 
+has the advantage of supporting complex types, and also has performance optimizations that are not present in Comet's 
+existing reader.
+
+Support should still be considered experimental, but most of Comet’s unit tests are now passing with the new reader. 
+Known issues include handling of `INT96` timestamps and unsigned bytes and shorts.
```

Comment:
We merged @mbutrovich 's fix for this into arrow this week, so it should be available shortly!